### PR TITLE
feat: do NOT load preserved catalogs on late DB creation

### PR DESCRIPTION
**Pull request by crepererum**
When a DB is created AFTER the server is initialized, then we can assume
it is a new DB (because the rules file did not exist beforehand). We
shall treat it as a new DB with no data and should not try to load some
leftover / stale / whatever preserved catalog for it. How this catalog
came into existence we do not know and it was certainly not properly
managed by IOx. So we error if there is a catalog.

Furthermore the old implementation was kinda broken since it loaded the
perserved catalog "in-sync" with the gRPC call that issued the DB
creation (we only have a delayed init concept for DBs that are loaded on
instance startup). In production that would very likely provoke nasty
timeouts.

On top of that this new behavior will also be somewhat more sane when we
think about sequencer (e.g. Kafka) replays. We certainly do not wanna do
any replays for newly created DBs.

TLDR: New behavior for DBs created via gRPC is "new empty DB". This does
NOT affect DBs loaded on instance startup (aka existing DBs).

Closes #1785.

### CODE REVIEW

**Comment by mkmik** [comment]
just to be sure I understand: if `create_preserved_catalog` fails, the DB will be marked as failed in the status API?

**Comment by crepererum** [comment]
> just to be sure I understand: if `create_preserved_catalog` fails, the DB will be marked as failed in the status API?

Then the gRPC call will return with an error and the DB will not be created. So errors in the status API can only happen due to errors during startup, not while the server is online. Hence #1785 is NOT required anymore since you don't need a per-DB `initialized` flag.

**Review by mkmik — APPROVED** [review]

**Review by domodwyer — APPROVED** [review]
